test/automation/src/sql/notebook.ts

```diff
@@ -362,15 +365,39 @@ export class NotebookView {
 	async searchInNotebook(expr: string): Promise<IElement> {
 		await this.waitForSetSearchValue(expr);
 		await this.code.dispatchKeybinding('enter');
-		let selector = `${NotebookView.actualResult} `;
+		let selector = NotebookView.searchResult;
 		if (expr) {
-			selector += '.message';
+			selector += ' .message';
 		}
-		return await this.code.waitForElement(selector, undefined);
+		return this.code.waitForElement(selector, undefined);
 	}
 
 	async waitForSetSearchValue(text: string): Promise<void> {
 		const textArea = `${NotebookView.inputBox} textarea`;
 		await this.code.waitForTypeInEditor(textArea, text);
 	}
+
+	/**
+	 * Helper function
+	 * @returns tree item ids from Notebooks View
+	 */
+	async getNotebookTreeItemIds(): Promise<string[]> {
+		return (await this.code.waitForElements(NotebookView.notebookTreeItem, false)).map(item => item.attributes['id']);
+	}
+
+	async pinNotebook(): Promise<void> {
+		await this.code.dispatchKeybinding('escape');
+		const notebookIds = await this.getNotebookTreeItemIds();
+		await this.code.waitAndDoubleClick(`${NotebookView.notebookTreeItem}[id="${notebookIds[0]}"]`);
```

Comment:
You should probably add a JS doc comment to the function and mention that currently it will pin the first notebook in the tree